zuul-core/src/main/java/com/netflix/zuul/message/Headers.java

```diff
@@ -548,19 +624,53 @@ private void truncate(int i) {
         }
     }
 
+    /**
+     * Checks if the given value is compliant with our RFC 7230 based check
+     */
+    private boolean isValid(String value) {
+        if (value != null) {
+            int pos = findInvalid(value);
+            if (pos != ABSENT) {
+                Spectator.globalRegistry().counter("zuul.header.invalid.char").increment();
+                return false;
+            }
+        }
+        return true;
```

Comment:
You could use Guard Clause this way we could have some code readability improvement here. 

```suggestion
        if (value == null) {
            return true;
        }
        if (findInvalid(value) == ABSENT) {
                return true;
        }
        Spectator.globalRegistry().counter("zuul.header.invalid.char").increment();
        return false;
```